Django 2.1.5 update

PaulLocksley opened the pull request:
Hi team,
 Django 2.0 had an issue with sqlite that prevented me from testing, quickly updated to 2.1.5 and have been able to use it so far without issue.

Let me know if you have any issues.

PaulLocksley committed:
Django 2.1.5 update

Razorfang commented:
Before we continue with this, have you found anything so far with this update, or with version 2.0?

PaulLocksley commented:
I've been able to do my user testing on this branch. 
Tested my usual upgrade checklist:
- Different users / institutions.
- xml generation.
- Map functions (both global and local items).
- General user behaviour works as I expected.

So far I've only found issues with invalid form tooltips formatting that exists in production.

Razorfang commented:
I had a look through the release notes for the in-between Django versions and some features that we're using are removed in Django 2.1.

In the file `djnro/urls.py`, we are using `django.contrib.auth.views.login`, and in the file `edumanage/views.py`, we are using `django.contrib.auth.views.logout`, both of which are removed according to 
https://docs.djangoproject.com/en/5.1/releases/2.1/#features-removed-in-2-1 .

Also, in `edumanage/views.py`, we use the `host` parameter of `is_safe_url`, which is also removed.

I don't think we can safely merge this change until these points are addressed.

PaulLocksley committed:
fix: update is_safe_url param

PaulLocksley commented:
I believe I replaced all the views login logout views. 
Nice catch on the is_safe_url, I've modified it to use the new parameters specified in the linked article.
When we get to django 3 we will need to change it again to url_has_allowed_host_and_scheme.

I did some quick testing with next url set to valid and invalid values ie:
http://localhost:3034/setlang/?next=https://google.com.au
http://localhost:3034/setlang/?next=/faq/en

And it seems to still work as expected.

Razorfang commented:
I'll rebase the changes with mine for now. I'm still working on fixing some of the required changes for user registration and activation. I'll open a pull request once everything is working as expected.

Razorfang commented:
Status update.

***@***.***

I am currently trying to fix an error that occurs for shibboleth based logins.

I am also not 100% sure that the registration and activation of new users fully works, but for now, it’s more important that I fix this login issue. That will at least provide a functional site for existing users.

Once these issues have been fixed and tested I’ll open a merge request for the full update from Django 1.11 to Django 2.1.5.


From: Paul ***@***.***>
Date: Monday, 20 January 2025 at 2:59 PM
To: REANNZ/djnro ***@***.***>
Cc: Jamie Getty ***@***.***>, Comment ***@***.***>
Subject: Re: [REANNZ/djnro] Django 2.1.5 update (PR #1)

I believe I replaced all the views login logout views.
Nice catch on the is_safe_url, I've modified it to use the new parameters specified in the linked article.
When we get to django 3 we will need to change it again to url_has_allowed_host_and_scheme.

I did some quick testing with next url set to valid and invalid values ie:
http://localhost:3034/setlang/?next=https://google.com.au
http://localhost:3034/setlang/?next=/faq/en

And it seems to still work as expected.

—
Reply to this email directly, view it on GitHub<https://github.com/REANNZ/djnro/pull/1#issuecomment-2601169493>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD454UMSJ3Y2WMFLPET3MIL2LRJ55AVCNFSM6AAAAABVLDKYOGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMMBRGE3DSNBZGM>.
You are receiving this because you commented.Message ID: ***@***.***>
